<?xml version="1.0"?>
<launch>

  <!-- Apriltag -->
  <group>
    <arg name="launch_prefix" default="" />

    <arg name="camera_name" default="/camera/rgb" />
    <arg name="image_topic" default="image_raw" />

    <arg name="queue_size" default="1" />

    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
      <!-- Remap topics from those used in code to those on the ROS network -->
      <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
      <remap from="camera_info" to="$(arg camera_name)/camera_info" />

      <!-- default: false -->
      <!-- <param name="publish_tag_detections_image" type="bool" value="true" /> -->

      <param name="queue_size" type="int" value="$(arg queue_size)" />

      <!-- load parameters (incl. tag family, tags, etc.) -->
      <rosparam command="load" file="$(find initial_pose_apriltag)/config/apriltag_settings.yaml"/>
      <rosparam command="load" file="$(find initial_pose_apriltag)/config/apriltag_tags.yaml"/>
    </node>
  </group>

  <!-- Show in Rviz   -->
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find coconut_docking)/rviz/docking_monitor.rviz" /> -->

  <!-- Show in Rqt   -->
  <!-- <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/> -->

</launch>
